### PR TITLE
chore: Refactor styled component to css modules for checkbox/radio/switch

### LINKED ISSUE #27386: Rewrite Radio to CSS modules

### LINKED ISSUE #27387: Rewrite Checkbox to CSS modules

### LINKED ISSUE #27388: Rewrite Switch to CSS modules

### PR DESCRIPTION
**Pull request by jsartisan**
Refactor css from styled component to css modules

Fixed #27386 
Fixed #27388
Fixed #27387

### CODE REVIEW

File: app/client/packages/design-system/headless/src/components/Checkbox/src/Checkbox.tsx [hunk 1]
```diff
@@ -86,6 +87,7 @@ const _Checkbox = (props: CheckboxProps, ref: CheckboxRef) => {
       data-hovered={isHovered ? "" : undefined}
       data-invalid={validationState === "invalid" ? "" : undefined}
       data-label=""
+      data-label-position={labelPosition}
```
Comment (jsartisan) [inline app/client/packages/design-system/headless/src/components/Checkbox/src/Checkbox.tsx:90]: note: this is required to position the inline label. Previously we were doing it with dynamic css.

File: app/client/packages/design-system/headless/src/components/Switch/Switch.tsx [hunk 2]
```diff
@@ -18,7 +18,6 @@ export interface SwitchProps
   extends Omit<SpectrumSwitchProps, keyof StyleProps>,
     Validation,
     InlineLabelProps {
-  icon?: React.ReactNode;
```
Comment (jsartisan) [inline app/client/packages/design-system/headless/src/components/Switch/Switch.tsx:21]: this was a miss on our end. this prop is not required.

File: app/client/packages/design-system/widgets/src/components/Checkbox/stories/Checkbox.stories.mdx [hunk 3]
```diff
@@ -124,7 +124,7 @@ Checkbox is a component that allows the user to select one or more options from
   name="Custom Icon"
   args={{
     children: "Custom Icon",
-    icon: <EmotionHappyLineIcon />,
+    icon: EmotionHappyLineIcon,
```
Comment (jsartisan) [inline app/client/packages/design-system/widgets/src/components/Checkbox/stories/Checkbox.stories.mdx:127]: This was a miss in my previous PR.

File: app/client/packages/design-system/widgets/src/styles/src/field.module.css [hunk 4]
```diff
@@ -41,12 +23,6 @@ export const fieldStyles = css<FieldStylesProps>`
     gap: var(--spacing-1);
     height: fit-content;
     color: var(--color-fg);
-    font-weight: ${({ isEmphasized }) => (isEmphasized ? "bold" : "normal")};
```
Comment (jsartisan) [inline app/client/packages/design-system/widgets/src/styles/src/field.module.css:44]: Note removing the code for the side label just for CSS, I'll raise another PR to remove the full support in types and component

File: app/client/packages/design-system/widgets/src/components/CheckboxGroup/stories/CheckboxGroup.stories.mdx [hunk 5]
```diff
@@ -47,30 +47,6 @@ Checkbox Group is a group of checkboxes that can be selected together.
...
->
-  {Template.bind({})}
-</Story>
-
-## Label Position and Alignment
-
-<Story
-  args={{
-    labelPosition: "side",
-    labelWidth: "100px",
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/CheckboxGroup/stories/CheckboxGroup.stories.mdx:66]: If we don't use such `labelPosition: "side"`, then we probably don't need `labelWidth` as a prop, right?

File: app/client/packages/design-system/headless/src/components/Checkbox/src/Checkbox.tsx (hunk 1 above)
Comment (KelvinOm) [reply app/client/packages/design-system/headless/src/components/Checkbox/src/Checkbox.tsx:90]: Seems like this doesn't work for switch. 
Sep-21-2023 12-24-00

Comment (KelvinOm) [reply app/client/packages/design-system/headless/src/components/Checkbox/src/Checkbox.tsx:90]: Also do we need the same option for the radio?

**Review by KelvinOm — CHANGES_REQUESTED** [review]
Please check the affected components once more time. For example, the input component has a broken label.

Снимок экрана 2023-09-21 в 12 26 33

File: app/client/packages/design-system/widgets/src/components/CheckboxGroup/stories/CheckboxGroup.stories.mdx (hunk 5 above)
Comment (jsartisan) [reply app/client/packages/design-system/widgets/src/components/CheckboxGroup/stories/CheckboxGroup.stories.mdx:66]: Right.

File: app/client/packages/design-system/headless/src/components/Checkbox/src/Checkbox.tsx (hunk 1 above)
Comment (KelvinOm) [reply app/client/packages/design-system/headless/src/components/Checkbox/src/Checkbox.tsx:90]: Fixed

File: app/client/packages/design-system/widgets/src/components/CheckboxGroup/stories/CheckboxGroup.stories.mdx (hunk 5 above)
Comment (KelvinOm) [reply app/client/packages/design-system/widgets/src/components/CheckboxGroup/stories/CheckboxGroup.stories.mdx:66]: Will be fixed [here](https://github.com/appsmithorg/appsmith/issues/27298).

**Review by KelvinOm — APPROVED** [review]

**Pull request merged**